src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/TensorPrimitives.Divide.cs

```diff
@@ -70,11 +72,103 @@ public static void Divide<T>(T x, ReadOnlySpan<T> y, Span<T> destination)
         internal readonly struct DivideOperator<T> : IBinaryOperator<T> where T : IDivisionOperators<T, T, T>
         {
             public static bool Vectorizable => typeof(T) == typeof(float)
-                                            || typeof(T) == typeof(double);
+                                            || typeof(T) == typeof(double)
+                                            || (Vector256.IsHardwareAccelerated && typeof(T) == typeof(int));
             public static T Invoke(T x, T y) => x / y;
-            public static Vector128<T> Invoke(Vector128<T> x, Vector128<T> y) => x / y;
-            public static Vector256<T> Invoke(Vector256<T> x, Vector256<T> y) => x / y;
-            public static Vector512<T> Invoke(Vector512<T> x, Vector512<T> y) => x / y;
+            public static Vector128<T> Invoke(Vector128<T> x, Vector128<T> y)
+            {
+                if (typeof(T) == typeof(int))
```

Comment:
The changes here look overall reasonable, but its still an area where we'd like to ensure its handling the right scenarios and applying everywhere, not just `TensorPrimitives`.

That is, this would ideally be getting done in `operator /` in Vector64/128/256/512. We'd ideally be using a throw helper so that we're not hurting inlinability. We'd ideally also be handling the other sizes (`byte`, `sbyte`, `short`, `ushort`, and `uint`) that can all use the same general algorithm.

The only case that can't be trivially handled is division for `long` and `ulong` as those require special handling since they might result in loss of precision when converted to `double`.

Reply:
Correct. In general we want support to be available to anyone doing vector division, not just to TensorPrimitives. By optimizing `operator /` on the Vector64/128/256/512 types instead, we ensure this happens. We then do the work in the JIT to ensure that users aren't pessimized for something that is going to generate somewhere between 1-5 instructions.